From: Abhijith Das <adas@redhat.com>
Date: Fri, 4 Jun 2010 16:37:23 -0400
Subject: [fs] gfs2: use -EUSERS when mounting w/o enough journals
Message-id: <1848453220.594411275669443196.JavaMail.root@zmail05.collab.prod.int.phx2.redhat.com>
Patchwork-id: 25977
O-Subject: [RHEL5.6 PATCH][GFS2] - Bug 600387 - gfs2 kernel - Better error
	reporting when mounting a gfs fs without enough journals
Bugzilla: 600387
RH-Acked-by: Steven Whitehouse <swhiteho@redhat.com>

Use -EUSERS to signal to mount helper about insufficient journals so it doesn't
display a cryptic message. This one-liner is upstream

Resolves: rhbz#600387
Signed-off-by: Abhi Das <adas@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/gfs2/ops_fstype.c b/fs/gfs2/ops_fstype.c
index d2e8206..75dbe03 100644
--- a/fs/gfs2/ops_fstype.c
+++ b/fs/gfs2/ops_fstype.c
@@ -661,7 +661,7 @@ static int init_journal(struct gfs2_sbd *sdp, int undo)
 		goto fail;
 	}
 
-	error = -EINVAL;
+	error = -EUSERS;
 	if (!gfs2_jindex_size(sdp)) {
 		fs_err(sdp, "no journals!\n");
 		goto fail_jindex;
